src/utils/createBadgeImage.ts

```diff
@@ -1,78 +1,36 @@
-import {
-  CompositeOperator,
-  Gravity,
-  IMagickImage,
-  MagickColors,
-  MagickGeometry,
-  MagickImage,
-  Point,
-} from '@imagemagick/magick-wasm';
+import { IMagickImage, MagickImage } from '@imagemagick/magick-wasm';
 
-import BadgeOptions from '../types/BadgeOptions';
+import BadgeOptions, { scaleBadgeOptions } from '../types/BadgeOptions';
+import TextOptions, { scaleTextOptions } from '../types/TextOptions';
 import drawCenteredText from './drawCenteredText';
+import getTextBoundingBox from './getTextBoundingBox';
+import roundToEven from './roundToEven';
 
 export default function createBadgeImage(
   badgeOptions: BadgeOptions,
-  expectedWidth: number,
+  textOptions: TextOptions,
+  scale: number,
 ): IMagickImage {
-  const baseHeight = 30;
-  const baseWidth = 130;
-  const baseShadowSize = 3;
-  const baseRatio = baseHeight / baseWidth;
+  const scaledBadgeOptions = scaleBadgeOptions(badgeOptions, scale);
+  const scaledTextOptions = scaleTextOptions(textOptions, scale);
 
-  // Scale the sizes up to the expected badge size if larger
-  const width = Math.max(baseWidth, expectedWidth);
-  const height = Math.round(width * baseRatio);
-  const scale = width / baseWidth;
-  const shadowSize = Math.floor(baseShadowSize * scale);
-  const fontPointSize = Math.floor(badgeOptions.text.fontPointSize * scale);
-
-  const heightWithShadow = height + shadowSize * 2;
-
-  // Create the shadow canvas filled with the shadow color
-  const shadow = MagickImage.create();
-  shadow.backgroundColor = MagickColors.None;
-  shadow.read(badgeOptions.shadowColor, width, height);
-
-  // Expand the canvas by the shadow size and blur by the shadow amount
-  shadow.extent(
-    new MagickGeometry(width, heightWithShadow),
-    Gravity.Center,
-    MagickColors.Transparent,
+  const textBox = getTextBoundingBox(
+    scaledTextOptions,
+    192 * scale,
+    192 * scale,
```

Comment:
Remind what the 192 here represents.

Reply:
In the calling code, we scale the sizes of things from a 192px icon base. By that I mean when you specify a font size of 32 it is for icons that are 192x192.  Any icons above or below that gets their sizes (font, padding, shadow, etc) scaled relative to it.

This particular use is just creating a blank canvas of the maximum expected size to render the text on so we can get the bounding box without guessing at a canvas size.  I can adjust it to pass in the size from the parent so we're not reversing a calculation done in the code calling this and can add a comment there explaining what 192 is being used there (which then should be the only use of it).